kernel/src/hil/time.rs

```diff
@@ -47,6 +47,29 @@ pub trait Ticks: Clone + Copy + From<u32> + fmt::Debug + Ord + PartialOrd + Eq {
 
     /// Returns the maximum value of this type, which should be (2^width)-1.
     fn max_value() -> Self;
+
+    /// Coverts the specified val into this type if it fits otherwise the
+    /// `max_value()` is returned
+    #[inline]
+    fn from_or_max(val: u64) -> Self {
+        if val < Self::max_value().into_u32() as u64 {
```

Comment:
I don't understand this part. This function is supposed to, for time `T: Ticks`, given a `u64`, construct a `T` with either the passed number or `<T as Ticks>::max_value()`, right?

Won't comparing with `Self::max_value().into_u32()` set the upper bound to a `u32` then?

Reply:
This provided implementation works for all `Ticks` expect `Ticks64`, which provides its own implementation. I could instead copy and paste this same code as the implementation for `Ticks16`, `Ticks24`, and `Ticks32` and not have the default implementation.

Even though I lean toward keeping this default implementation for everything but `Ticks64`, I can remove it in favor of every `Ticks` implementing explicitly. Just let me know.

Reply:
Can we please either add a warning comment that this auto-implemented function will not work for any Ticks > 32-bit, or just not provide an auto-implementation?

I have a bad feeling about this. An auto-implementation should always work if it's there. Not implementing this on a custom `Ticks` type is going to cause a ton of headaches. Also, people may easily miss the Ticks64 implementation of this method and assume wrong behavior, given that other Ticks don't have their own implementation of this.

A clearly visible comment might be fine, if an explicit implementation per `impl Ticks` feels to verbose for you.

Reply:
Done. I just implemented in every type. This way there are not future landmine (even though having another ticks greater than 32 is very unlikely).